src/cascadia/TerminalSettingsEditor/ColorSchemeViewModel.cpp

```diff
@@ -0,0 +1,137 @@
+                        else if (const auto stringTag{ tag.try_as<hstring>() })
+                        {
+                            if (stringTag == ForegroundColorTag)
+                            {
+                                _scheme.Foreground(newColor);
+                            }
+                            else if (stringTag == BackgroundColorTag)
+                            {
+                                _scheme.Background(newColor);
+                            }
+                            else if (stringTag == CursorColorTag)
+                            {
+                                _scheme.CursorColor(newColor);
+                            }
+                            else if (stringTag == SelectionBackgroundColorTag)
+                            {
+                                _scheme.SelectionBackground(newColor);
+                            }
+                        }
+                    }
+                }
+            }
+        };
+
+        for (uint8_t i = 0; i < ColorTableSize; ++i)
+        {
+            til::color currentColor{ scheme.Table()[i] };
+            const auto& entry{ winrt::make<ColorTableEntry>(i, currentColor) };
+            entry.PropertyChanged(colorEntryChangedHandler);
+            if (i < ColorTableDivider)
+            {
+                _NonBrightColorTable.Append(entry);
+            }
+            else
+            {
+                _BrightColorTable.Append(entry);
+            }
+        }
+
+        _ForegroundColor = winrt::make<ColorTableEntry>(ForegroundColorTag, til::color(scheme.Foreground()));
+        _BackgroundColor = winrt::make<ColorTableEntry>(BackgroundColorTag, til::color(scheme.Background()));
+        _CursorColor = winrt::make<ColorTableEntry>(CursorColorTag, til::color(scheme.CursorColor()));
+        _SelectionBackgroundColor = winrt::make<ColorTableEntry>(SelectionBackgroundColorTag, til::color(scheme.SelectionBackground()));
+
+        _ForegroundColor.PropertyChanged(colorEntryChangedHandler);
+        _BackgroundColor.PropertyChanged(colorEntryChangedHandler);
+        _CursorColor.PropertyChanged(colorEntryChangedHandler);
+        _SelectionBackgroundColor.PropertyChanged(colorEntryChangedHandler);
+    }
+
+    winrt::hstring ColorSchemeViewModel::Name()
+    {
+        return _Name;
+    }
+
+    void ColorSchemeViewModel::Name(winrt::hstring newName)
+    {
+        _scheme.Name(newName);
+        _Name = newName;
+    }
+
+    Editor::ColorTableEntry ColorSchemeViewModel::ColorEntryAt(uint32_t index)
+    {
+        if (index < ColorTableDivider)
+        {
+            return _NonBrightColorTable.GetAt(index);
+        }
+        else
+        {
+            return _BrightColorTable.GetAt(index - ColorTableDivider);
+        }
+    }
+
+    ColorTableEntry::ColorTableEntry(uint8_t index, Windows::UI::Color color)
+    {
+        static const std::array<hstring, 16> TableColorNames = {
+            RS_(L"ColorScheme_Black/Text"),

```

Comment:
actually... because you're looking these up entirely at runtime, you _don't_ need to change the localization keys...